Added support for header-only style and preinstalled volk through CMake config packages.

jweg opened the pull request:
As discussed extensively in the previous PR.

Quite a handful of files changed, so here's a quick overview:
- volk.c and generate.py haven't changed.
- volk.h only adds the optional include of volk.c
- CMakeLists.txt has support for config installation (as non-default behaviour, to enable add_subdirectory without polluting the user project with install files)
- the rest is 3 minimal test applications integrating volk in various ways through CMake, and the travis config for building all of them.

zeux commented:
Would it be possible to keep the static library target, perhaps using smth like VOLK_HEADER parameter? As it stands this makes it impossible to just add volk as a CMake dependency and have it work - as your CMake example with add_subdirectory shows, you have to add VOLK_IMPLEMENTATION to one of your source files.

jweg commented:
Yes, that's possible. We could additionally have what we had before, a "volk" static lib target, just with a way to add defines for build time.
The explicit-include way is a tad more powerful: it allows setting the define (i.e. deciding the platform integration method) not only at buildsystem level, but also at compile time. Also installing volk prior to project build time (vcpkg again) really only makes sense with the header-only style. But in the interest of backwards-compatibility we can nevertheless have both ways coexist.
I've made those changes in the branch and improved the tests to actually not compile when the platform defines were not passed through properly. Also I grew to dislike the parsing of volk.h in CMakeLists (having the version is valueable, though, requiring a min version of the vulkan headers in find_package(volk) actually works), so I changed generate.py to also configure the CMakeLists.txt. 

"The CMake files now build two targets: volk and volk_headers. The first is a static library where platform defines can be set using VOLK_STATIC_DEFINES, the latter is for the header-only approach.

- Refactored the tests to test all relevant situations, and added testing for whether setting the platform defines actually worked.
- Added a bash script to run the test cases locally. The travis script now just calls that script.
- Changed the Vulkan header version detection in CMakeLists.txt to be more in line with the volk approach: it is now written at generate time just like the volk sources.
- Fixed a bug from trunk that could break the Win32 build (missing defined() in volk.h:27)"

zeux commented:
Thank you! This looks great - it's a bit more complex, but there's two cleanly separated targets so that users can pick & choose whatever works best for them. Appreciate the perseverance :)
